plugins/libplugin.c

```diff
@@ -2115,8 +2115,14 @@ static struct listpeers_channel *json_to_listpeers_channel(const tal_t *ctx,
 
 	json_to_node_id(buffer, idtok, &chan->id);
 	json_to_bool(buffer, conntok, &chan->connected);
-	json_to_bool(buffer, privtok, &chan->private);
+	if (privtok)
+		json_to_bool(buffer, privtok, &chan->private);
+	else
+		chan->private = true;
 	chan->state = json_strdup(chan, buffer, statetok);
+	if (streq(chan->state, "DUALOPEND_OPEN_INIT") ||
+	    streq(chan->state, "OPENINGD"))
+		return tal_free(chan);
 	json_to_txid(buffer, ftxidtok, &chan->funding_txid);
```

Comment:
No, just hoist the scid and alias handling up (or, move all this down below it).

That correctly handles the "no scid, ignore channel" case, which *also* fixes this, without making a dependency on the state names.

Reply:
Good point! Fixed.